Test/domains/urcu-bp.hpp

```diff
@@ -3,22 +3,23 @@
 #include "rcu_domain.hpp"
 
 #include <urcu-bp.h>
-#define _LGPL_SOURCE
 
-namespace std {
-    class rcu_bp: public rcu_domain {
-    public:
-        void register_thread() { rcu_register_thread(); }
-        void unregister_thread() { rcu_unregister_thread(); }
-        void read_lock() noexcept { rcu_read_lock(); }
-        void read_unlock() noexcept { rcu_read_unlock(); }
-        void synchronize() noexcept { synchronize_rcu(); }
-        void retire(rcu_head *rhp, void (*cbf)(rcu_head *rhp)) { call_rcu(rhp, cbf); }
-        void barrier() noexcept { rcu_barrier(); }
-        void quiescent_state() noexcept { rcu_quiescent_state(); }
-        void thread_offline() noexcept { rcu_thread_offline(); }
-        void thread_online() noexcept { rcu_thread_online(); }
+class rcu_domain_bp {
+public:
+    static constexpr bool register_thread_needed() { return false; }
+    void register_thread() { rcu_register_thread(); }
+    void unregister_thread() { rcu_unregister_thread(); }
+    void thread_offline() noexcept { rcu_thread_offline(); }
+    void thread_online() noexcept { rcu_thread_online(); }
 
-        static constexpr bool register_thread_needed() { return false; }
-    };
-} // namespace std
+    static constexpr bool quiescent_state_needed() { return false; }
+    void quiescent_state() noexcept { rcu_quiescent_state(); }
```

Comment:
Yes, if the RCU domain knows for sure that a given function is a no-op, then it should be plenty OK to avoid calling it.

And yes, I was in fact missing the bit about a sane default being supplied.  I don't see a problem with that approach.